tests/lib/screenshot-testing/support/page-renderer.js

```diff
@@ -486,13 +513,26 @@ PageRenderer.prototype._getImageLoadingCount = function () {
 };
 
 PageRenderer.prototype._waitForNextEvent = function (events, callback, i, waitTime) {
+
+    function isEmpty(obj) {
+        for (var key in obj) {
+            if (Object.prototype.hasOwnProperty.call(obj, key)) return false;
+        }
+
+        return true;
+    }
+
     var self = this;
     setTimeout(function () {
-        if (self._getAjaxRequestCount() == 0
-            && self._getImageLoadingCount() == 0
-            && !self._isLoading
-        ) {
-            self._executeEvents(events, callback, i + 1);
+        if (!self._isLoading && !self._isInitializing && !self._isNavigationRequested) {
+            var hasPhantomNoPendingRequests = isEmpty(self._resourcesRequested)
+
+            if (hasPhantomNoPendingRequests || (!self._getAjaxRequestCount() && !self._getImageLoadingCount())) {
+                // why isEmpty(self._resourcesRequested) || !self._getAjaxRequestCount()) ?
+                // if someone sends a sync XHR we only get a resoruceRequested event but not a responseEvent so we need to
+                // fall back for ajaxRequestCount as a safety net. See https://github.com/ariya/phantomjs/issues/11284
+                self._executeEvents(events, callback, i + 1);
```

Comment:
There's an issue here, since it's now a nested if, if the second condition is false, the function just exists. So _waitForNextEvent isn't called again in that case.